config.c

```diff
@@ -696,30 +696,16 @@ static char *_get_value_policy(struct dl_list *config_list)
 	return strdup(item);
 }
 
-static int _apply_config_sysctl(const char *key, const char *value,
-				void *opaque)
+static int _set_config_sysctl_by_key(const char *key, const char *value,
+				     void *opaque)
 {
-	const char *start = key + strlen("sysctl");
-	char *path =
-		calloc(strlen("/proc/sys") + strlen(start) + 1, sizeof(char));
-
+	char *path = pv_config_parser_sysctl_key(key);
 	if (!path)
 		return -1;
 
-	sprintf(path, "%s", "/proc/sys");
-
-	const char *next = start;
-	char *p = path + strlen("/proc/sys");
-
-	for (int i = 0; i < (int)strlen(start); ++i)
-		p[i] = next[i] == '.' ? '/' : next[i];
-
-	errno = 0;
 	int fd = open(path, O_WRONLY | O_SYNC);
 	if (fd < 0) {
-		pv_log(ERROR, "open failed for sysctl node %s with '%s'", path,
-		       strerror(errno));
-
+		pv_log(ERROR, "cannot open '%s': %s", path, strerror(errno));
```

Comment:
since we seem to ignore the errors in the iterator, maybe we should make this a WARN?